apps/runtime/src/main/java/quarks/apps/runtime/MonitorApp.java

```diff
@@ -0,0 +1,204 @@
+    private final Topology topology;
+    private static final Logger logger = LoggerFactory.getLogger(MonitorApp.class);
+
+    /**
+     * Constructs a {@code MonitorApp} with the specified name in the 
+     * context of the specified provider.
+     * 
+     * @param provider the topology provider
+     * @param submitter a {@code DirectSubmitter} which provides required 
+     *      services and submits the application
+     * @param name the application name
+     * 
+     * @throws IllegalArgumentException if the submitter does not provide 
+     *      access to the required services
+     */
+    public MonitorApp(TopologyProvider provider, 
+            DirectSubmitter<Topology, Job> submitter, String name) {
+
+        this.provider = provider;
+        this.submitter = submitter;
+        validateSubmitter();
+        this.topology = declareTopology(name);
+    }
+    
+    /**
+     * Submits the application topology.
+     * 
+     * @return the job.
+     * @throws InterruptedException
+     * @throws ExecutionException
+     */
+    public Job submit() throws InterruptedException, ExecutionException {
+        Future<Job> f = submitter.submit(topology);
+        return f.get();
+    }
+
+    /**
+     * Submits an application using an {@code ApplicationServiceMXBean} control 
+     * registered with the specified {@code ControlService}.
+     * 
+     * @param applicationName the name of the application to submit
+     * @param controlService the control service
+     */
+    public static void submitApplication(String applicationName, ControlService controlService) {
+        try {
+            Set<ApplicationServiceMXBean> controls = 
+                    controlService.getControls(ApplicationServiceMXBean.class);
+            if (controls.isEmpty()) {
+                throw new IllegalStateException(
+                        "Could not find a registered control with the following interface: " + 
+                        ApplicationServiceMXBean.class.getName());                
+            }
+            for (ApplicationServiceMXBean control : controls)
+// TODO add ability to submit with the initial application configuration
+                control.submit(applicationName, null);
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    
+    /**
+     * Declares the following topology:
+     * <pre>
+     * JobEvents source --> Filter (health == unhealthy) --> Restart application
+     * </pre>
+     * 
+     * @param name the topology name
+     * @return the application topology
+     */
+    protected Topology declareTopology(String name) {
+        Topology t = provider.newTopology(name);
+        TStream<JsonObject> jobEvents = JobEvents.source(
+                t, 
+                (evType, job) -> { return MonitorAppEvent.toJsonObject(evType, job); }
+                );
+        jobEvents = PlumbingStreams.isolate(jobEvents, true);
```

Comment:
Isn't the isolate superfluous?  JobEvents.source() just wraps Topology.source() and that defines the stream is isolated.  Probably good to add the same doc to JobEvents.source().